src/service/alerts/alert.rs

```diff
@@ -555,19 +556,19 @@ pub async fn delete_by_name(
     stream_name: &str,
     name: &str,
 ) -> Result<(), AlertError> {
-    if db::alerts::alert::get_by_name(org_id, stream_type, stream_name, name)
-        .await
-        .is_err()
-    {
+    let Some(alert) =
+        db::alerts::alert::get_by_name(org_id, stream_type, stream_name, name).await?
+    else {
         return Err(AlertError::AlertNotFound);
+    };
+
+    db::alerts::alert::delete_by_name(org_id, stream_type, stream_name, name).await?;
+    remove_ownership(org_id, "alerts", Authz::new(name)).await;
+    if let Some(alert_id) = alert.id.map(|id| id.to_string()) {
+        remove_ownership(org_id, "alerts", Authz::new(&alert_id)).await;
     }
-    match db::alerts::alert::delete_by_name(org_id, stream_type, stream_name, name).await {
-        Ok(_) => {
-            remove_ownership(org_id, "alerts", Authz::new(name)).await;
-            Ok(())
-        }
-        Err(e) => Err(e.into()),
-    }
+
+    Ok(())
 }
```

Comment:
The old alert routes will add/remove ownership using both the old `name` identifier and the new `id` identifier.

We are using the old `name` identifier so that the alerts which are created deleted **before** the `alerts_openfga_id` migration is ran and **before** the application switches over to using the V2 routes still have the old `name` identifier in OpenFGA and so that RBAC for those alerts still functions.

We are using the new `id` identifier so that alerts which are created/deleted **after** the `alerts_openfga_id` migration is ran but **before** the application switches over to using the V2 routes still have the new `id` identifier in OpenFGA without requiring that the client runs the `alerts_openfga_id` migration a second time.